contrib/android/fetch_barcode_scanner.sh

```diff
@@ -0,0 +1,56 @@
+#!/bin/bash
+
+# script to fetch and pin https://github.com/markusfisch/BarcodeScannerView and its dependencies,
+# https://github.com/markusfisch/CameraView/ and https://github.com/markusfisch/zxing-cpp
+# which are being used as barcode scanner in the Android app.
+
+# To bump the version of BarcodeScannerView, get the newest version tag from the github repo,
+# then get the required dependencies from
+# https://jitpack.io/com/github/markusfisch/BarcodeScannerView/**NEWEST_VERSION**/BarcodeScannerView-**NEWEST_VERSION**.pom
+# then fetch the aars from jitpack and update the versions and sha256s below. Also update kotlin-stdlib
+# in buildozer_qml.spec
+
+BARCODE_SCANNER_VIEW_VERSION="1.6.0"
+BARCODE_SCANNER_VIEW_AAR_SHA256="2be6c9a5ab86f7198683af4a6c0e5acd3e8fe6a02df2d12c3b716dc422537789"
+
+CAMERA_VIEW_VERSION="1.9.2"
+CAMERA_VIEW_AAR_SHA256="3c9be35d29b84637d2a2b0e0e7253bc5a35408fafb26c5cb7225aeb7326e2be4"
+
+ZXING_CPP_VERSION="v2.2.0.1"
+ZXING_CPP_AAR_SHA256="7991381f181ff16555c4ac9c5d83e6a0d3a7da896efb8c3807897305ca33b957"
+
+DOWNLOAD_REPOSITORY_ROOT="https://jitpack.io/com/github/markusfisch"
+
+set -e
+
+CONTRIB_ANDROID="$(dirname "$(readlink -e "$0")")"
+CONTRIB="$CONTRIB_ANDROID"/..
+
+. "$CONTRIB"/build_tools_util.sh
+
+# check if contrib/android/aars exists and contains BarcodeScannerView.aar, CameraView.aar and zxing-cpp.aar already,
+# if not, create and fetch them
+if [ ! -d "$CONTRIB_ANDROID/aars" ]; then
+    mkdir -p "$CONTRIB_ANDROID/aars"
+fi
+
+if [ ! -f "$CONTRIB_ANDROID/aars/BarcodeScannerView.aar" ]; then
+    info "Fetching BarcodeScannerView..."
+    curl -L "${DOWNLOAD_REPOSITORY_ROOT}/BarcodeScannerView/${BARCODE_SCANNER_VIEW_VERSION}/BarcodeScannerView-${BARCODE_SCANNER_VIEW_VERSION}.aar" \
+        -o "$CONTRIB_ANDROID/aars/BarcodeScannerView.aar"
+fi
+echo "$BARCODE_SCANNER_VIEW_AAR_SHA256  $CONTRIB_ANDROID/aars/BarcodeScannerView.aar" | sha256sum -c -
```

Comment:
Could this be replaced with `download_if_not_exist` and `verify_hash` calls?
See e.g.
https://github.com/spesmilo/electrum/blob/26e6c0900ba4f3bc10a25e0df814141fe937cfef/contrib/build-linux/appimage/make_appimage.sh#L46-L47